### ISSUE TITLE
[Bug]: No error when starting the app with location turned off.

### ISSUE THREAD

**Issue by natsuk4ze** [Baseflow/flutter-geolocator#1408]
### Please check the following before submitting a new issue.

- [X] I have searched the [existing issues](https://github.com/baseflow/flutter-geolocator/issues).
- [X] I have carefully [read the documentation](https://github.com/Baseflow/flutter-geolocator/blob/main/geolocator/README.md) and verified I have added the required platform specific configuration.

### Please select affected platform(s)

- [X] Android
- [ ] iOS
- [ ] Linux
- [ ] macOS
- [ ] Web
- [ ] Windows

### Steps to reproduce

1. Start debugging with sample code below in Android
2. Reject location dialog from system

📝 Note:
> As far as I have been able to find, there is no way to completely turn off the Android emulator's location services. Even if you turn off GPS from the settings, after a while you will still get location information using wifi as the provider, so I suggest you try it on an actual Android device.

### Expected results

An error should occur because the developer is prompting the user to allow location services.

### Actual results

No error. In the sample code, the data is left `waiting`.
`getServiceStatusStream` does not work during that time either.

### Code sample

<details><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:geolocator/geolocator.dart';

void main() => runApp(const App());

class App extends StatelessWidget {
  const App({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: Center(
          child: StreamBuilder<Position>(
            stream: Geolocator.getPositionStream(),
            builder: (context, snapshot) {
              if (snapshot.hasError) return Text('Error: ${snapshot.error}');
              if (!snapshot.hasData) return const CircularProgressIndicator();

              final position = snapshot.data!;
              return Text('${position.latitude}, ${position.longitude}');
            },
          ),
        ),
      ),
    );
  }
}

```

</details>

### Screenshots or video

<details>
<summary>Screenshots or video demonstration</summary>

https://github.com/Baseflow/flutter-geolocator/assets/50433979/4bc8fc3c-fad9-4bea-919f-b29508d69210

</details>

### Version

10.1.0

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel beta, 3.19.0-0.1.pre, on macOS 14.2.1 23C71 darwin-arm64, locale ja-JP)
    • Flutter version 3.19.0-0.1.pre on channel beta at /Users/xxx/Hack/Tools/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 1a9a60d044 (4 days ago), 2024-01-10 13:38:09 -0800
    • Engine revision bbebee1465
    • Dart version 3.3.0 (build 3.3.0-279.0.dev)
    • DevTools version 2.31.0

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/xxx/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15C500b
    • CocoaPods version 1.14.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] VS Code (version 1.85.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.80.0

[✓] Connected device (4 available)
    • A063 (mobile)               • 192.168.1.6:42633 • android-arm64  • Android 13 (API 33)
    • sdk gphone64 arm64 (mobile) • emulator-5554     • android-arm64  • Android 14 (API 34) (emulator)
    • macOS (desktop)             • macos             • darwin-arm64   • macOS 14.2.1 23C71 darwin-arm64
    • Chrome (web)                • chrome            • web-javascript • Google Chrome 120.0.6099.216
    ! Error: Browsing on the local area network for iPhone (6). Ensure the device is unlocked and attached with a cable or associated with the same local area network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)
    ! Error: Browsing on the local area network for iPhone 13 mini. Ensure the device is unlocked and attached with a cable or associated with the same local area network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

**Comment by mvanbeusekom** [comment]
Hi @natsuk4ze,

Thank you for reporting this issue and providing the reproducing example. Thanks to this I was able to locate the bug and submit PR #1466 with a fix. 

Once the PR is approved, I will make sure to land it and publish the update.

**Comment by natsuk4ze** [comment]
Excellent. Thank you for always maintaining the project😌

**Comment by mvanbeusekom** [comment]
Hi @natsuk4ze, 

We have just release version 4.5.4 of the `geolocator_android` package containing a bug fix for this issue. If the application depends on the `geolocator` package, running `flutter pub upgrade` should download the updated version and resolve the issue. 

Thanks again for pointing out the issue.
